src/hooks/Rules.py

```diff
@@ -24,3 +27,72 @@ def anyClassLevel(world: World, mw: MultiWorld, state: CollectionState, player:
+# OptOne check if the passed item (with or without ||) is enabled, then return |item:count| where count is clamped to the maximum number of said item
+def OptOne(base: World, world: MultiWorld, state: CollectionState, player: int, item: str, items_counts: Optional[dict] = None):
+    """Returns item with count adjusted to Real Item Count"""
+    if item == "":
+        return "" #Skip this function if item is left blank
+    if not items_counts:
+        items_counts = _getItem_counts(base, world, player)
+
+    require_type = 'item'
+
+    if '@' in item[:2]:
+        require_type = 'category'
+
+    item = item.lstrip('|@$').rstrip('|')
+
+    item_parts = item.split(":")
+    item_name = item
+    item_count = 1
+
+    if len(item_parts) > 1:
+        item_name = item_parts[0]
+        item_count = item_parts[1]
+
+    if require_type == 'category':
+        if isinstance(item_count, int):
+            #Only loop if we can use the result to clamp
+            category_items = [item for item in base.item_name_to_item.values() if "category" in item and item_name in item["category"]]
+            category_items_counts = sum([items_counts.get(category_item["name"], 0) for category_item in category_items])
+            item_count = clamp(item_count, 0, category_items_counts)
+        return f"|@{item_name}:{item_count}|"
+    elif require_type == 'item':
+        if isinstance(item_count, int):
+            item_current_count = items_counts.get(item_name, 0)
+            item_count = clamp(item_count, 0, item_current_count)
+        return f"|{item_name}:{item_count}|"
+
+# OptAll check the passed require string and loop every item to check if they're enabled,
+# then returns the require string with counts ajusted using OptOne
+def OptAll(base: World, world: MultiWorld, state: CollectionState, player: int, requires: str):
+    """Returns an entire require string with counts adjusted to Real Item Count"""
+    requires_list = requires
+
+    items_counts = _getItem_counts(base, world, player)
+
+    functions = {}
+    if requires_list == "":
+        return True
+    for item in re.findall(r'\{(\w+)\(([^)]*)\)\}', requires_list):
+        #so this function doesnt try to get item from other functions, in theory.
+        func_name = item[0]
+        functions[func_name] = item[1]
+        requires_list = requires_list.replace("{" + func_name + "(" + item[1] + ")}", "{" + func_name + "(temp)}")
+    # parse user written statement into list of each item
+    for item in re.findall(r'\|[^|]+\|', requires):
+        itemScanned = OptOne(base, world, state, player, item, items_counts)
+        requires_list = requires_list.replace(item, itemScanned)
+
+    for function in functions:
+        requires_list = requires_list.replace("{" + function + "(temp)}", "{" + func_name + "(" + functions[func_name] + ")}")
+    return requires_list
+
+def _getItem_counts(base: World, world: MultiWorld, player: int):
```

Comment:
This function's contents is a duplicate of the same logic in the world's Rules file above: https://github.com/ManualForArchipelago/Manual/pull/23/files#diff-a25c303033861df67335b2a78c209196808d8d93e059588997e9f5422bad35e8L74-L79

Since both places need it, I'd rather see it as a public `get_item_counts` method on the world that needs no arguments, then both places can just call that getter. Or any hooks that might want to call it can call it there too. 😃